dev/io.openliberty.data.internal_fat/fat/src/test/jakarta/data/DataTest.java

```diff
@@ -37,13 +42,24 @@
 @MinimumJavaLevel(javaLevel = 11) // TODO 17
 public class DataTest extends FATServletClient {
 
+    @ClassRule
+    public static final JdbcDatabaseContainer<?> testContainer = DatabaseContainerFactory.create();
+
     @Server("io.openliberty.data.internal.fat")
     @TestServlets({ @TestServlet(servlet = DataTestServlet.class, contextRoot = "DataTestApp"),
                     @TestServlet(servlet = ProviderTestServlet.class, contextRoot = "ProviderTestApp") })
     public static LibertyServer server;
 
     @BeforeClass
     public static void setUp() throws Exception {
+        // Get driver type
+        DatabaseContainerType type = DatabaseContainerType.valueOf(testContainer);
+        server.addEnvVar("DB_DRIVER", type.getDriverName());
+        server.addEnvVar("DB_USER", testContainer.getUsername());
+        server.addEnvVar("DB_PASSWORD", testContainer.getPassword());
+
+        // Set up server DataSource properties
+        DatabaseContainerUtil.setupDataSourceProperties(server, testContainer);
```

Comment:
There are two setup method options: 

```java
DatabaseContainerUtil.setupDataSourceProperties(server, testContainer);
DatabaseContainerUtil.setupDataSourceDatabaseProperties(server, testContainer);
```

The first one will replace your `<properties.derby.embedded ... />` with `<properties .../>`.  This is for tests that want to specifically test generic properties. 

The second one will replace your `<properties.derby.embedded .../>` with `<properties.[the database being tested] .../>`.  

Either should work, but I figured I would point it out just in case you didn't know. 
(the method names are admittedly a bit confusing)